client/network/sync/src/lib.rs

```diff
@@ -1729,12 +1734,7 @@ where
 			let middle = best_seens.len() / 2;
 
 			// Not the "perfect median" when we have an even number of peers.
-			let median = *best_seens.select_nth_unstable(middle).1;
-			if median > self.best_queued_number {
-				Some(median)
-			} else {
-				None
-			}
```

Comment:
This reverts: https://github.com/paritytech/substrate/pull/12114

Reply:
As far as I can see, we would run into the same problem with your code.

Reply:
I think you meant #11817, right? If so then answer to my previous question about `SyncState::Importing` was no, it shouldn't return `Some(target)`, I'll revert it back to the previous logic as of #11817.